jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -112,16 +122,23 @@ def __init__(self, *args, **kwargs):
         # Custom dataset filters for Cubeviz
         if self.config == "cubeviz":
             def valid_cubeviz_datasets(data):
+
                 comp = data.get_component(data.main_components[0])
                 img_unit = u.Unit(comp.units) if comp.units else u.dimensionless_unscaled
+                solid_angle_unit = check_if_unit_is_per_solid_angle(img_unit, return_unit=True)
+                if solid_angle_unit is None:  # this is encountered sometimes ??

```

Comment:
This is the known? issue with _get_display_unit. When you start up the app, sometimes it will prematurely return a flux unit or None when requesting surface brightness. I didn't figure it was worth it to investigate since hopefully use of _get_display_unit will be cut down once we make changes to the plugins to respond directly to GlobalDisplayUnitChanged messages. Everything works out in the end, but this check needs to be here to avoid some of the weirdness when data is being loaded etc (cc @kecnry)

Reply:
we might be able to remove this in #3192 - if you know a case to check, can you try?  Otherwise I can investigate when I get a chance